src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp

```diff
@@ -1128,22 +1142,17 @@ void MultiDiracDeterminant::mw_updateRatios(const size_t det_offset,
   auto* ratios_list_ptr             = ratios_deviceptr_list.data();
   const auto* sign_ptr              = sign.data();
   const int* data_ptr               = data.data();
-  const auto* det0_list_ptr         = det0_list.data();
   const auto* table_matrix_list_ptr = table_matrix_deviceptr_list.data();
 
-  PRAGMA_OFFLOAD("omp target teams distribute parallel for collapse(2) map(always, to: det0_list_ptr[:nw])")
+  PRAGMA_OFFLOAD("omp target teams distribute parallel for collapse(2)")
   for (size_t iw = 0; iw < nw; iw++)
     for (size_t count = 0; count < ndet_ext; ++count)
     {
-      size_t det_id = det_offset + count;
-      ValueType ratios_local;
-      ///Initialization here to avoid one additional transfer and allow the use of collapse(2)
-      ratios_list_ptr[iw][0] = det0_list_ptr[iw];
```

Comment:
This was the bug and my original comment explains part of the cause.
mw_updateRatios<EXT_LEVEL> is only called when there is excitation.
Simply move the assignment out.